roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -650,14 +650,12 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
 	term->add(*pdf);
 	termNormDeps->add(pdfNormDeps, kFALSE);
 	termAllDeps->add(pdfAllDeps, kFALSE);
-	if (!termIntDeps) {
-	  termIntDeps = new RooArgSet("termIntDeps");
+	if (termIntDeps) {
+	  termIntDeps->add(*pdfIntSet, kFALSE);
```

Comment:
The code seems to me different now. If in the term loop , we never create here a`termIntDeps` set and a `termIntNoNormDeps` set.  Am I missing something ?

Reply:
True, in this loop it never gets created, but there is also an outer loop in which it might get created (line 679 and 680 in the current master code).

Reply:
ok, I understand it now. The `termIntDeps` set and `termIntNoNormDeps` are used only at lines 685 and 686 (old code), and this is executed only when the code above is not. SO there is no need to create them before. 
I think then we can merge now the PR
Thank you !